src/store/assets/actions.ts

```diff
@@ -1,47 +1,149 @@
+  const evmNetworkId = rootState.web3.evmNetworkSelected;
+  const networkAssets = await evmBridgeApi.getNetworkAssets(evmNetworkId as number);
+
+  const networkAssetsWithBalance = await Promise.all(
+    Object.entries(networkAssets).map(async ([soraAddress, assetData]) => {
+      const accountAsset = {
+        address: assetData.evmAddress as string,
+        balance: ZeroStringValue,
+        contract: assetData.contract,
+        decimals: 18,
+      };
+
+      try {
+        const { value, decimals } = await ethersUtil.getAccountAssetBalance(accountAddress, accountAsset.address);
+
+        accountAsset.balance = value;
+        accountAsset.decimals = decimals;
+
+        // update evmBalance
+        if (ethersUtil.isNativeEvmTokenAddress(accountAsset.address)) {
+          rootCommit.web3.setEvmBalance(value);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+
+      return {
+        [soraAddress]: accountAsset,
+      };
+    })
+  );
+
+  return networkAssetsWithBalance;
+}
 
 const actions = defineActions({
-  async updateRegisteredAssets(context, reset?: boolean): Promise<void> {
-    const { state, commit, rootDispatch } = assetsActionContext(context);
+  // for common usage
+  async updateRegisteredAssets(context, force?: boolean): Promise<void> {
+    const { state, commit, dispatch, rootState } = assetsActionContext(context);
 
-    if (state.registeredAssetsFetching) return;
+    if (state.registeredAssetsFetching) {
+      if (force) {
+        await delay(250);
+        return await dispatch.updateRegisteredAssets(force);
+      } else {
+        return;
+      }
+    }
 
-    if (reset) commit.resetRegisteredAssets();
+    if (force) {
+      commit.resetRegisteredAssets();
+    }
+
+    const networkType = rootState.web3.networkType;
 
     commit.setRegisteredAssetsFetching(true);
 
     try {
-      const registeredAssets = await bridgeApi.getRegisteredAssets();
-
-      const preparedRegisteredAssets = await Promise.all(
-        registeredAssets.map(async (item) => {
-          const accountAsset = { ...item, externalBalance: ZeroStringValue };
-
-          try {
-            if (!accountAsset.externalAddress) {
-              accountAsset.externalAddress = await rootDispatch.web3.getEvmTokenAddressByAssetId(item.address);
-            }
-            if (accountAsset.externalAddress) {
-              const { value, decimals } = await rootDispatch.web3.getBalanceByEvmAddress(accountAsset.externalAddress);
-              accountAsset.externalBalance = value;
-
-              if (!accountAsset.externalDecimals) {
-                accountAsset.externalDecimals = decimals;
-              }
-            }
-          } catch (error) {
-            console.error(error);
+      const networkAssetsWithBalance = await (async () => {
+        switch (networkType) {
+          case BridgeType.ETH: {
+            return await getEthRegisteredAssets(context);
```

Comment:
you can remove `await` here and `async` above cuz if you have Promises here - you have `await` above so you don't need it here